envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -9,19 +9,20 @@ import pl.allegro.tech.servicemesh.envoycontrol.services.LocalityAwareServicesSt
 import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
 import reactor.core.scheduler.Scheduler
-import java.util.function.BiFunction
 
 class SnapshotUpdater(
     private val cache: SnapshotCache<Group>,
     private val properties: SnapshotProperties,
     private val scheduler: Scheduler,
-    private val onGroupAdded: Flux<out Any>,
+    private val onGroupAdded: Flux<List<Group>>,
```

Comment:
I do not get why it was used here. `out` modifier is a `variance annotation` so it indicates that this can be a flux of anything that is a subtype of `Any` but by definition everything is a subtype of `Any`. Maybe the original author can explain that.

Reply:
Not really, it could maybe be `Flux<*>` but otherwise `Flux<out Any>` was needed to allow any subtype of Any, as `Flux<T>` is saying that you expect Flux of exactly that T type, so you could not pass `Flux<String>` to it.

In same way now it is impossible to pass like `Flux<ArrayList<Group>>` - but I guess it's not really needed to support such types (and I didn't analyze this changes, so not sure even if this is possible here for sure) in internal code, but i would say it's more valid according to producer extends consumer super rule.

It's easier to forget about this in kotin because many kotlin build in types have variance directly on type, like `List<out T>` for default immutable list, so you don't need to manually specify this for every variable/parameter/field.